code/modules/client/preferences_ui.dm

```diff
@@ -619,6 +627,25 @@
 
 				current_client.update_movement_keys()
 
+		if("setCustomSentence")
+			var/kb_name = params["name"]
+			if(!kb_name)
+				return
+			var/list/part = splittext(kb_name, ":")
+			var/id = text2num(part[2])
+			var/datum/custom_emote/emote = custom_emotes[id]
+			emote.message = params["sentence"]
+			custom_emotes[id] = emote
+
+		if("setEmoteType")
+			var/kb_name = params["name"]
+			if(!kb_name)
+				return
+			var/list/part = splittext(kb_name, ":")
+			var/id = text2num(part[2])
+			var/datum/custom_emote/emote = custom_emotes[id]
```

Comment:
is there no limit to the length of the emote?

Reply:
300 , it's in the TGUI